app/src/main/java/org/wikipedia/settings/SettingsActivity.java

```diff
@@ -23,12 +34,29 @@ public SettingsFragment createFragment() {
         return SettingsFragment.newInstance();
     }
 
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        initialLanguageList = StringUtil.listToJsonArrayString(app.language().getAppLanguageCodes());
+        initialFeedCardsEnabled = Prefs.getFeedCardsEnabled();
+        initialFeedCardsOrder = Prefs.getFeedCardsOrder();
+    }
+
     @Override
     public void onActivityResult(int requestCode, int resultCode, Intent data) {
         super.onActivityResult(requestCode, resultCode, data);
-        if (requestCode == ACTIVITY_REQUEST_ADD_A_LANGUAGE) {
+
+        String finalLanguageList = StringUtil.listToJsonArrayString(app.language().getAppLanguageCodes());
+        List<Boolean> finalFeedCardsEnabled = Prefs.getFeedCardsEnabled();
+        List<Integer> finalFeedCardsOrder = Prefs.getFeedCardsOrder();
```

Comment:
Would it be possible to directly use these the `Prefs.getFeedCardsEnabled()` and `Prefs.getFeedCardsOrder()` in the `if` statements instead of creating new variables?

Reply:
Yes we could do that, but the code would be less readable. I think keeping these variables makes the code more consistent as we have `finalLanguageList` too. Are you concerned about the performance impact?

Reply:
Although the performance impact might be really small, the `Prefs.getFeedCardsOrder()` and `Prefs.getFeedCardsEnabled()` have enough information from the names that indicate the values are the "final" or "current" data, which can compare with the "initial" ones.